ros_cross_compile/ros_cross_compile.py

```diff
@@ -139,16 +147,27 @@ def cross_compile_pipeline(
         default_docker_dir=sysroot_build_context,
         colcon_defaults_file=args.colcon_defaults)
 
-    gather_rosdeps(
-        docker_client=docker_client,
-        platform=platform,
-        workspace=ros_workspace_dir,
-        custom_script=custom_rosdep_script,
-        custom_data_dir=custom_data_dir)
+    if not args.skip_rosdep_collection:
+        gather_rosdeps(
+            docker_client=docker_client,
+            platform=platform,
+            workspace=ros_workspace_dir,
+            custom_script=custom_rosdep_script,
+            custom_data_dir=custom_data_dir)
+    check_install_script_exists(ros_workspace_dir, platform)
     create_workspace_sysroot_image(docker_client, platform)
     run_emulated_docker_build(docker_client, platform, ros_workspace_dir)
 
 
+def check_install_script_exists(ros_workspace_dir, platform):
```

Comment:
This helper script should live in `dependencies.py`.

Also, please add type annotations to the function declaration, e.g.

```
def assert_install_script_exists(ros_workspace_dir: Path, platform: Platform) -> bool:
```